src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -286,88 +286,22 @@ protected override void DeserializeCore(XmlElement element, SaveContext context)
-
-                            return new Sequence(startToken, new DoubleToken(1), endToken, convertToken) as IDoubleSequence;
-                        }
-                        else
-                        {
-                            IDoubleInputToken endToken = ParseToken(rangeIdentifiers[1], idSet, identifiers);
-
-                            if (rangeIdentifiers.Length > 2)
-                            {
-                                if (rangeIdentifiers[2].StartsWith("#"))
-                                {
-                                    var count = rangeIdentifiers[2].Substring(1);
-                                    IDoubleInputToken countToken = ParseToken(count, idSet, identifiers);
-
-                                    return new CountRange(startToken, countToken, endToken, convertToken);
-                                }
-
-                                if (rangeIdentifiers[2].StartsWith("~"))
-                                {
-                                    var approx = rangeIdentifiers[2].Substring(1);
-                                    IDoubleInputToken approxToken = ParseToken(approx, idSet, identifiers);
-
-                                    return new ApproxRange(startToken, approxToken, endToken, convertToken);
-                                }
-
-                                return new Range(startToken, ParseToken(rangeIdentifiers[2], idSet, identifiers), endToken, convertToken);
-                            }
-
-                            double identifierValue0, identifierValue1;
-                            var canBeParsed0 = System.Double.TryParse(rangeIdentifiers[0], out identifierValue0);
-                            var canBeParsed1 = System.Double.TryParse(rangeIdentifiers[1], out identifierValue1);
-
-                            //both of the value can be parsed as double
-                            if (canBeParsed0 && canBeParsed1)
-                            {
-                                if (identifierValue0 < identifierValue1)
-                                    return new Range(startToken, new DoubleToken(1), endToken, convertToken) as IDoubleSequence;
-                                else
-                                    return new Range(startToken, new DoubleToken(-1), endToken, convertToken) as IDoubleSequence;
-                            }
-
-                            //the input cannot be parsed as double, return a default function and let it handle the error
-                            return new Range(startToken, new DoubleToken(1), endToken, convertToken) as IDoubleSequence;
-                        }
-
-                    }
+                    IDoubleInputToken startToken = ParseToken(x, idSet, identifiers);
 
                     return new OneNumber(startToken, convertToken) as IDoubleSequence;
                 }).ToList();
+
+            if (doubleSequenceList.Count == 0)
+                throw new Exception("No identifiers found.");
```

Comment:
When is this exception thrown, when the number node is empty? Can you reproduce it? If so, please add the error message as string resource.

Reply:
Yes it was supposed to cover that case but I just checked and it is already being checked somewhere else. I will delete those lines